js/wizard.js

```diff
@@ -0,0 +1,437 @@
+      ethConfig += "      nameservers:\r\n"
+      ethConfig += "        addresses: ["
+      for(var i = 0; i < dns_server_list.length; i++) {
+        if (i > 0) {
+          ethConfig += ","
+        }
+        ethConfig += dns_server_list[i];
+      }
+      ethConfig += "]\r\n";
+    }
+
+    v2Str += ethConfig;
+
+    v2Str += "    all-eth:\r\n";
+    v2Str += "      match:\r\n";
+    v2Str += "        name: \"eth*\"\r\n";
+    v2Str += ethConfig;
+  }
+
+  if(has_wifi == true) {
+
+    v2Str += "  wifis:\r\n";
+    v2Str += "    all-wlans:\r\n";
+    v2Str += "      match:\r\n";
+    v2Str += "        name: \"wl*\"\r\n";  // This will match both old (wlan0) and newer (wlp7s0) naming schemas
+
+    v2Str += "      access-points:\r\n";
+    v2Str += "        \"" + wifi_ssid + "\":\r\n";
+
+    if(wifi_passphrase != "" || wifi_hidden_ssid) {
+      v2Str += "          password: \"" + wifi_passphrase + "\"\r\n";
+    }
+
+    if(wifi_dhcp == true) {
+      v2Str += "      dhcp4: true\r\n";
+    }
+    else {
+      v2Str += "      addresses: [" + wifi_ip + "/" + convertNetmaskToCIDR(wifi_mask) + "]\r\n";
+      v2Str += "      gateway4: " + wifi_gw + "\r\n";
+    }
+
+    if(has_custom_dns == true) {
+      v2Str += "      nameservers:\r\n"
+      v2Str += "        addresses: ["
+      for(var i = 0; i < dns_server_list.length; i++) {
+        if (i > 0) {
+          v2Str += ","
+        }
+        v2Str += dns_server_list[i];
+      }
+      v2Str += "]\r\n";
+    }
+  }
+
+  var validation_errors_length = validation_errors.length;
+  for (var i = 0; i < validation_errors_length; i++) {
+    if(validation_errors[i].key == 'common_error')
+    {
+      $('#common_errors').show();
+      $('#common_errors').append('<p class="help-block">' + validation_errors[i].message + '</p>');
+    }
+    $('#' + validation_errors[i].key).closest('.form-group').addClass("has-error");
+    if(validation_errors[i].key.startsWith("dns_server_") || validation_errors[i].key.startsWith("ntp_server_"))
+    {
+      $('#' + validation_errors[i].key).closest('.form-group').append('<p class="col-xs-offset-3 col-md-offset-3 col-xs-6 help-block">' + validation_errors[i].message + '</p>');
+    }
+    else {
+      $('#' + validation_errors[i].key).closest('.form-group').append('<p class="col-md-offset-3 col-xs-6 help-block">' + validation_errors[i].message + '</p>');
+    }
+
+  }
+
+  if(validation_errors_length == 0)
+  {
+    if(v1Str.length > 5)
+    {
+      if($('input[id=screenlyv1]').is(':checked'))
+      {
+        var configStr = v1Str;
+        var configFileName = "network.ini"
+      }
+      else
+      {
+        var configStr = v2Str;
+        var configFileName = "network.yaml"
+      }
+      alert("Your config file is almost ready!\n\nYour browser will start to download the file as soon as you press OK below. Save the file on to the boot partition of your SD card and make sure to name it " + configFileName + "\n\n");
```

Comment:
Since we are using Bootstrap why not use the [modal component](http://getbootstrap.com/docs/3.3/javascript/#modals)?